agent/adapter.cpp

```diff
@@ -82,7 +82,7 @@ void Adapter::setAgent(Agent &agent)
 
 void Adapter::addDevice(string &device)
 {
-	Device *dev = m_agent->getDeviceByName(device);
+	auto dev = m_agent->getDeviceByName(device);
```

Comment:
These changes look trivial but do allow subsequent refactoring to occur without changing large amounts of code.

For instance: Suppose we changed the return type of 'getDeviceByName' from 'Device*' to 'std::shared_ptr<Device>'. As this does not semantically change the type's behaviour, this code would not require change.

'auto' should be encouraged almost everywhere.